### PR TITLE
AGL US 11-12 Tests

### PR DESCRIPTION
**Pull request by alex-latham**
* Reorganizes users/new_spec.rb to use setup for user_info.
* Adds tests to ensure email address is unique and form fields are filled out completely.
* Fixes merchant_spec test which failed occasionally due to order of non-ordered array elements.
* Includes migration file for "users" table

Note: two tests fail (the new ones) - this PR was not an attempt to create passing user stories, just to have the tests added for when we continue development together.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -67,6 +67,18 @@
     t.index ["item_id"], name: "index_reviews_on_item_id"
   end
 
+  create_table "users", force: :cascade do |t|
```
Comment (tylertomlinson) [inline db/schema.rb:70]: **AR Enum**
we can add a column with "role" have the default: 0 
0 - Visitor
1 -  Regular User 
2 - Merchant Employee
3 - Admin User

File: spec/features/users/new_spec.rb [hunk 2]
```diff
@@ -2,27 +2,96 @@
...
+          expect(page).to have_content("New Registration Form")
+
+          fill_in :name, with: @user_info1[:name]
+          fill_in :address,	with: @user_info1[:address]
+          fill_in :city, with: @user_info1[:city]
+          fill_in :state,	with: @user_info1[:state]
+          fill_in :zip,	with: @user_info1[:zip]
+          fill_in :email,	with: @user_info1[:email]
+          fill_in :password, with: @user_info1[:password]
+          fill_in :password_confirmation,	with: @user_info1[:password]
```
Comment (tylertomlinson) [inline spec/features/users/new_spec.rb:37]: Is this spacing between symbol and with shown on the text editor?

File: spec/features/users/new_spec.rb [hunk 3]
```diff
@@ -2,27 +2,96 @@
...
+          click_button "Submit"
+
+          expect(current_path).to eq("/profile")
+          expect(page).to have_content("Welcome #{@user_info1[:name]}! You are now registered and logged in!")
         end
+
+        it "cannot create a new profile with an email address already in use" do
```
Comment (tylertomlinson) [inline spec/features/users/new_spec.rb:44]: I think we can test this on the model level

File: spec/features/users/new_spec.rb [hunk 4]
```diff
@@ -2,27 +2,96 @@
...
+          fill_in :zip,	with: @user_info1[:zip]
+          fill_in :email,	with: @user_info1[:email]
+          fill_in :password, with: @user_info1[:password]
+          fill_in :password_confirmation,	with: @user_info1[:password]
+          click_button "Submit"
+
+          expect(current_path).to eq("/profile")
+          expect(page).to have_content("Welcome #{@user_info1[:name]}! You are now registered and logged in!")
         end
+
```
Comment (tylertomlinson) [inline spec/features/users/new_spec.rb:43]: We can also test duplicate emails on the model level

File: spec/features/users/new_spec.rb (hunk 2 above)
Comment (alex-latham) [reply spec/features/users/new_spec.rb:37]: I noticed it, too, but they are regular single-spaces (not tabs) in my editor.

File: spec/features/users/new_spec.rb (hunk 4 above)
Comment (alex-latham) [reply spec/features/users/new_spec.rb:43]: Oh, I forgot to consider that!

**Comment by tylertomlinson** [comment]
Looks good

**Review by tylertomlinson — APPROVED** [review]
Looks good

**Review by danielpselph — APPROVED** [review]
Changes look good, approve changes.
